dags/miovision_check.py

```diff
@@ -0,0 +1,100 @@
+r"""### Daily Miovision Data Check DAG
+Pipeline to run additional SQL data quality checks on daily miovision pull.
+Put 'yellow card' checks which would not warrant the re-running of the data_pull pipeline,
+ie. issues which suggest field maintenance of sensors required. 
+"""
+import sys
+import os
+
+from airflow.decorators import dag
+from datetime import timedelta
+from airflow.models import Variable 
+from airflow.sensors.external_task import ExternalTaskSensor
+
+import logging
+import pendulum
+
+try:
+    repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+    sys.path.insert(0, repo_path)
+    from dags.dag_functions import task_fail_slack_alert
+    from dags.custom_operators import SQLCheckOperatorWithReturnValue
+except:
+    raise ImportError("Cannot import DAG helper functions.")
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+DAG_NAME = 'miovision_check'
+DAG_OWNERS = Variable.get('dag_owners', deserialize_json=True).get(DAG_NAME, ["Unknown"])
+
+default_args = {
+    'owner': ','.join(DAG_OWNERS),
+    'depends_on_past':False,
+    'start_date': pendulum.datetime(2023, 12, 21, tz="America/Toronto"),
+    'email_on_failure': False,
+    'email_on_success': False,
+    'retries': 0,
+    'retry_delay': timedelta(minutes=5),
+    'on_failure_callback': task_fail_slack_alert
+}
+
+@dag(
+    dag_id=DAG_NAME,
+    default_args=default_args,
+    schedule='0 4 * * *', # Run at 4 AM local time every day
+    catchup=False,
+    tags=["miovision", "data_checks"],
+    doc_md=__doc__
+)
+def miovision_check_dag():
+
+    t_upstream_done = ExternalTaskSensor(
+        task_id="starting_point",
+        external_dag_id="pull_miovision",
+        external_task_id="done",
```

Comment:
I think this is supposed to reference the task_id aka `pull_miovision` once more. DAG name and task name should really be distinct, I'm glad we are changing the dag name in another dag :meow_dio:

Reply:
nvm I saw your externaltaskmarker :meow_dio: